sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -279,9 +280,15 @@ case class HashAggregateExec(
     .map(_.asInstanceOf[DeclarativeAggregate])
   private val bufferSchema = StructType.fromAttributes(aggregateBufferAttributes)
 
-  // The name for Vectorized HashMap
-  private var vectorizedHashMapTerm: String = _
-  private var isVectorizedHashMapEnabled: Boolean = _
+  // The name for Fast HashMap
+  private var fastHashMapTerm: String = _
+  // whether vectorized hashmap or row based hashmap is enabled
+  // we make sure that at most one of the two flags is true
+  // i.e., assertFalse(isVectorizedHashMapEnabled && isRowBasedHashMapEnabled)
+  private var isVectorizedHashMapEnabled: Boolean = false
+  private var isRowBasedHashMapEnabled: Boolean = false
+  // auxiliary flag, true if any of two above is true
+  private var isFastHashMapEnabled: Boolean = false
```

Comment:
This flag check if one of them is enabled. As some of the generated code is same for both hash maps, the flag could make condition-checking clearer.